logstream/logprocessor.go

```diff
@@ -104,3 +103,51 @@ func GetRegexMatchingProcessor(logStream *LogStream, pattern string) (*LogProces
 
 	return logProcessor, processFn, nil
 }
+
+func createTemporarySnapshot(file *os.File) (*os.File, error) {
+	// Duplicate the file descriptor (so that when we work with the file, we don't affect the original file descriptor, especially cursor position)
+	fd, err := syscall.Dup(int(file.Fd()))
+	if err != nil {
+		return nil, err
+	}
+	readFile := os.NewFile(uintptr(fd), "name_doesnt_matter.txt")
```

Comment:
nit: This filename will be confusing to me if I catch it. Best to name it something like `snapshot.txt`